studio-api/components/WebServer/routecontrollers/conversation/generations.js

```diff
@@ -0,0 +1,260 @@
+  if (!jobId || !process.env.LLM_GATEWAY_SERVICES) return false
+
+  try {
+    const jobStatus = await llm.getJobStatus(jobId)
+    return jobStatus !== null
+  } catch (err) {
+    if (err.response?.status === 404) {
+      return false
+    }
+    // For other errors, assume job might exist
+    appLogger.warn(`[Generations] Error verifying job ${jobId}: ${err.message}`)
+    return true
+  }
+}
+
+/**
+ * List generations for a conversation and service
+ * GET /conversations/:conversationId/generations?serviceId=xxx
+ *
+ * Validates each completed generation against LLM Gateway and filters out orphans.
+ */
+async function listGenerations(req, res, next) {
+  try {
+    if (!req.params.conversationId) throw new ConversationIdRequire()
+
+    const { serviceId } = req.query
+    if (!serviceId) {
+      throw new ConversationMetadataRequire("serviceId query parameter is required")
+    }
+
+    // Verify conversation exists
+    const conversation = await model.conversations.getById(req.params.conversationId)
+    if (conversation.length !== 1) throw new ConversationNotFound()
+
+    // Get generations for this conversation and service
+    const generations = await model.conversationGenerations.listByConversationAndService(
+      req.params.conversationId,
+      serviceId
+    )
+
+    // Validate completed generations against LLM Gateway and filter out orphans
+    const validGenerations = []
+    for (const gen of generations) {
+      // Only validate completed jobs with jobId
+      if (gen.status === "completed" && gen.jobId) {
+        const jobExists = await verifyJobExists(gen.jobId)
+        if (!jobExists) {
+          // Check gateway health before deleting orphan
+          const gatewayHealthy = await checkLlmGatewayHealth()
+          if (gatewayHealthy) {
+            // Job no longer exists on gateway - delete orphan reference
+            appLogger.info(`[Generations] Removing orphan generation for job ${gen.jobId}`)
+            await model.conversationGenerations.delete(gen._id)
+            continue // Skip this generation
+          }
+          // Gateway unhealthy - keep reference, include in list
+        }
+      }
+      validGenerations.push(gen)
+    }
+
+    // Format response
+    const formattedGenerations = validGenerations.map((gen) => ({
+      generationId: gen.generationId,
+      jobId: gen.jobId,
+      createdAt: gen.createdAt,
+      serviceId: gen.serviceId,
+      serviceName: gen.serviceName || "",
+      status: gen.status,
+      isCurrent: gen.isCurrent,
+    }))
+
+    res.status(200).send({ generations: formattedGenerations })
+  } catch (error) {
+    appLogger.error(`[Generations] listGenerations error: ${error.message}`)
+    next(error)
+  }
+}
+
+/**
+ * Create a new generation record
+ * POST /conversations/:conversationId/generations
+ * Body: { serviceId: string, jobId: string, serviceName?: string }
+ */
+async function createGeneration(req, res, next) {
```

Comment:
Refactor `studio-api/components/WebServer/routecontrollers/conversation/generations.js` to behave like a proxy, limiting responsibilities to only what is required.  
Move the use of `executeBeforeResult` into this proxy logic.  

Reference similar logic in:  
- `components/WebServer/routes/router.js` (around line 177: `createProxyRoutes`)  
- `components/WebServer/routes/proxy/sessions/session.js`